src/net/socket.rs

```diff
@@ -116,8 +159,51 @@ pub fn bind_unix<Fd: AsFd>(sockfd: &Fd, addr: &SocketAddrUnix) -> io::Result<()>
     imp::net::syscalls::bind_unix(sockfd, addr)
 }
 
+/// `connect(sockfd, addr)`—Initiates a connection to an IP address.
+///
+/// # References
+///  - [POSIX]
+///  - [Linux]
+///
+/// [POSIX]: https://pubs.opengroup.org/onlinepubs/9699919799/functions/connect.html
+/// [Linux]: https://man7.org/linux/man-pages/man2/connect.2.html
+pub fn connect<Fd: AsFd>(sockfd: &Fd, addr: &SocketAddr) -> io::Result<()> {
+    let sockfd = sockfd.as_fd();
+    _connect(sockfd, addr)
+}
+
+fn _connect(sockfd: BorrowedFd<'_>, addr: &SocketAddr) -> io::Result<()> {
+    match addr {
+        SocketAddr::V4(v4) => imp::net::syscalls::connect_v4(sockfd, v4),
+        SocketAddr::V6(v6) => imp::net::syscalls::connect_v6(sockfd, v6),
+    }
+}
+
+/// `connect(sockfd, addr)`—Initiates a connection.
+///
+/// # References
+///  - [POSIX]
+///  - [Linux]
+///
+/// [POSIX]: https://pubs.opengroup.org/onlinepubs/9699919799/functions/connect.html
+/// [Linux]: https://man7.org/linux/man-pages/man2/connect.2.html
+#[doc(alias = "connect")]
+pub fn connect_any<Fd: AsFd>(sockfd: &Fd, addr: &SocketAddrAny) -> io::Result<()> {
+    let sockfd = sockfd.as_fd();
+    _connect_any(sockfd, addr)
```

Comment:
Early on, I had wanted to minimize monomorphization, and converting from `AsFd` types into concrete `BorrowedFd` types as early as possible seemed like it might help do that. But in practice it doesn't seem to make much of a difference.

Reply:
I've now added a patch converting all the code to the form you suggest.

Reply:
I think for the cases where you've already split out a separate function, there cannot be any difference at all. 

BTW, there is https://docs.rs/momo/0.2.2/momo/ - never tried it myself though.

Reply:
The cases I was thinking about were things like `mkdirat`, where the fd gets captured by a lambda. But looking at it now, it doesn't seem to matter much there either.

momo is cool, and I sometimes wish something like that were built into Rust itself, so that any function that wants a `&str` can take an `AsRef<str>`, automatically. But as things are today, I don't want rustix to have a proc_macro dependency.